manifests/init.pp

```diff
@@ -865,26 +956,69 @@
 
       case $::osfamily {
         'RedHat': {
+          case $::operatingsystemmajrelease {
+            '6': {
 
-          file { 'pam_system_auth_ac':
-            ensure  => file,
-            path    => $system_auth_ac_file,
-            content => template('pam/system-auth-ac.erb'),
-            owner   => 'root',
-            group   => 'root',
-            mode    => '0644',
-            require => Package[$my_package_name],
-          }
+              file { 'pam_system_auth_ac':
+                ensure  => file,
+                path    => $system_auth_ac_file,
+                content => template('pam/system-auth-ac.erb'),
+                owner   => 'root',
+                group   => 'root',
+                mode    => '0644',
+                require => Package[$my_package_name],
+              }
 
-          file { 'pam_system_auth':
-            ensure  => symlink,
-            path    => $system_auth_file,
-            target  => $system_auth_ac_file,
-            owner   => 'root',
-            group   => 'root',
-            require => Package[$my_package_name],
-          }
+              file { 'pam_system_auth':
+                ensure  => symlink,
+                path    => $system_auth_file,
+                target  => $system_auth_ac_file,
+                owner   => 'root',
+                group   => 'root',
+                require => Package[$my_package_name],
+              }
 
+              file { 'pam_password_auth_ac':
+                ensure  => file,
+                path    => $password_auth_ac_file,
+                content => template('pam/password-auth-ac.erb'),
+                owner   => 'root',
+                group   => 'root',
+                mode    => '0644',
+                require => Package[$my_package_name],
+              }
+
+              file { 'pam_password_auth':
+                ensure  => symlink,
+                path    => $password_auth_file,
+                target  => $password_auth_ac_file,
+                owner   => 'root',
+                group   => 'root',
+                require => Package[$my_package_name],
+              }
+            }
+            default: {
+
+              file { 'pam_system_auth_ac':
```

Comment:
instead of duplicating the default in the case statement for EL6, these could live outside of the case, so that they always apply to RedHat regardless of the version.